lib/plug/conn.ex

```diff
@@ -1320,10 +1320,19 @@ defmodule Plug.Conn do
   `get_http_protocol/1` to retrieve the protocol and version.
   """
   @spec inform(t, status, Keyword.t()) :: t
-  def inform(%Conn{} = conn, status, headers \\ []) do
+  def inform(%Conn{adapter: {adapter, _}} = conn, status, headers \\ []) do
     status_code = Plug.Conn.Status.code(status)
-    adapter_inform(conn, status_code, headers)
-    conn
+
+    case adapter_inform(conn, status_code, headers) do
+      :ok ->
+        conn
+
+      {:ok, payload} ->
+        put_in(conn.adapter, {adapter, payload})
+
+      {:error, :not_supported} ->
+        conn
```

Comment:
I can add a catch-all clause for backwards compatibility, let me know. If so perhaps with a warning?

```suggestion
      {:error, :not_supported} ->
        conn

      _other ->
        conn
```

there's no need for a catch-all clause in inform!/3 as it already was rasing.